### PR TITLE
added array functions

### PR DESCRIPTION
**Pull request by KalyaniPayyavula**
- Add array functions
- Add unit tests

### CODE REVIEW

File: learnArrays/learnArrays.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    let maxNum = Number.NEGATIVE_INFINITY
+    testArr.forEach(element => {
+        if (element > maxNum) {
+            maxNum = element
+        }
+    });
+    return maxNum;
+}
+
+export let findCommonEle = (firstArr, secondArr) => {
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: Change name of the function. There is one more function with the name `findCommonElem`

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]
Fix all review comments.

File: learnArrays/learnArrays.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+// function to find maximum number in the array
+// Args : array of numbers
+
+export let findMax = (testArr) => { // [2,3,4]
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: Why did you use `let` here? why not `const` ? Same applies for other functions as well.

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: learnArrays/learnArrays.js (hunk 1 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: fixed

File: learnArrays/learnArrays.js (hunk 2 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: fixed

File: learnArrays/learnArrays.test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    expect(removeDuplicateNumbers(1)).toBe("Arguments should be arrays")
+})
+
+test("find common elements", () => {
+    expect(findCommonElem([1, 2, 3], [3, 4])).toEqual([3])
+})
+
+test("find most repeated number", () => {
+    expect(mostRepNum([1, 2, 2])).toEqual(["2"]);
+    expect(mostRepNum([1, 3, 2, 2, 3, 3, 2])).toEqual(["2", "3"])
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.test.js]: I think better way of doing this is to use another method .. https://jestjs.io/docs/expect#expectarraycontainingarray

File: learnArrays/learnArrays.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    if (!Array.isArray(arrOne) || !Array.isArray(arrTwo)) {
+        return "input should be arrays"
+    }
+    if (arrOne.length == 0 || arrTwo.length == 0) {
+        return "Arrays should not be empty"
+    }
+    let combinedArray = [...arrOne, ...arrTwo]
+    let combUniqArr = []
+    combinedArray.forEach(ele => {
+        (combUniqArr.indexOf(ele) == -1) ? combUniqArr.push(ele) : ""
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: why do we need ternary operator here ? why not if?

File: learnArrays/learnArrays.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    })
+    let finRes = combUniqArr.filter(ele => ele > 3)
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: Method name says `greaterThanTwo` but condition shows >3 why ? Why `let` here ?

File: learnArrays/learnArrays.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+// Find numbers greater 2 in the given 2 arrays(no duplicates) using filter
+
+export const greaterThanTwo = (arrOne, arrTwo) => {
+    if (!Array.isArray(arrOne) || !Array.isArray(arrTwo)) {
+        return "input should be arrays"
+    }
+    if (arrOne.length == 0 || arrTwo.length == 0) {
+        return "Arrays should not be empty"
+    }
+    let combinedArray = [...arrOne, ...arrTwo]
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: Why not `const` ?

File: learnArrays/learnArrays.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        }
+    });
+    return maxNum;
+}
+
+export const removeDuplicateNumbers = (firstArr, secondArr) => {
+    if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+        return "Arguments should be arrays"
+    }
+    let combinedArr = [...firstArr, ...secondArr]
```
Comment (yugandhar-pathi) [inline learnArrays/learnArrays.js]: `const` here?

**Review by yugandhar-pathi — CHANGES_REQUESTED** [review]

File: learnArrays/learnArrays.test.js (hunk 3 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.test.js]: done

File: learnArrays/learnArrays.js (hunk 4 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: changed

File: learnArrays/learnArrays.js (hunk 5 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: changed

File: learnArrays/learnArrays.js (hunk 6 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: changed

File: learnArrays/learnArrays.js (hunk 7 above)
Comment (KalyaniPayyavula) [reply learnArrays/learnArrays.js]: changed

**Review by yugandhar-pathi — APPROVED** [review]
Good Job!! LGTM. Approving PR

**Pull request merged**